### PR TITLE
Convert Dia as a float field to a Link field

### LINKED ISSUE #68: Need to change dia attribute values for template in install.py

### LINKED ISSUE #69: Need to delete int conversion of cutting dia BOM.

### LINKED ISSUE #71: Need to change the Dia Fixed values in all template

### LINKED ISSUE #73: Dia converted into data field

### PR DESCRIPTION
**Pull request by Vigneshsekar**
Fixes #73 
Fixes #71 
Fixes #69 
Fixes #68 

If this PR is getting merged, we need to manually alter the Dia in the templates.

### CODE REVIEW

**Review by KaviyaPeriyasamy — CHANGES_REQUESTED** [review]
Need to change the dia attribute in the roll printing template.

File: apparelo/apparelo/doctype/apparelo_dia/apparelo_dia.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+		for value in item_attribute.item_attribute_values:
+			dias.append(value.attribute_value)
+		
+		if not self.dia in dias:
+			item_attribute.append('item_attribute_values',{
+				"attribute_value" : self.dia,
+				"abbr" : self.dia
+			})
+		item_attribute.save()
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/apparelo_dia/apparelo_dia.py]: push this into the if condition... do this for all other doctypes too...

File: apparelo/apparelo/doctype/cutting/cutting.py [hunk 2]
```diff
@@ -103,10 +103,7 @@ def get_attribute_values(self, attribute_name, part=None):
 						attribute_value.add(colour_mapping.colour)
 		elif attribute_name == "Dia":
 			for detail in self.details:
-				if int(str(float(detail.dia)).split('.')[1]) > 0:
-					attribute_value.add(str(detail.dia))
-				else:
-					attribute_value.add(str(detail.dia).split('.')[0])
+				attribute_value.add(str(detail.dia))
```
Comment (karthikeyan5) [inline apparelo/apparelo/doctype/cutting/cutting.py]: remove `str()`

File: apparelo/apparelo/patches/v1/dia_to_apparelo_dia.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	frappe.db.sql("""update `tabKnitting Dia` set dia = trim(trailing '0' from dia);""")
+	frappe.db.sql("""update `tabKnitting Dia` set dia = trim(trailing '.' from dia);""")
+
+	frappe.db.sql("""update `tabSteaming Dia Conversion` set from_dia = trim(trailing '0' from from_dia);""")
+	frappe.db.sql("""update `tabSteaming Dia Conversion` set from_dia = trim(trailing '.' from from_dia);""")
+
+	frappe.db.sql("""update `tabSteaming Dia Conversion` set to_dia = trim(trailing '0' from to_dia);""")
+	frappe.db.sql("""update `tabSteaming Dia Conversion` set to_dia = trim(trailing '.' from to_dia);""")
+
+	frappe.db.commit()
```
Comment (karthikeyan5) [inline apparelo/apparelo/patches/v1/dia_to_apparelo_dia.py]: remove `db.commit` ?

**Review by karthikeyan5 — CHANGES_REQUESTED** [review]

**Review by karthikeyan5 — APPROVED** [review]

**Pull request merged**